tools/k8s_versions/update_k8s_versions.go

```diff
@@ -0,0 +1,294 @@
+// the sorted ValidKubernetesVersions slice from constants_kubernetes_versions.go
+func getLatestSupportedMinikubeVersions(url string, k8sVersions []KubernetesVersion) ([]string, error) {
+	body, err := getRequestBody(url)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get minikube versions: %w", err)
+	}
+
+	// Extract the slice using a regular expression
+	re := regexp.MustCompile(`ValidKubernetesVersions = \[\]string{([^}]*)}`)
+	matches := re.FindStringSubmatch(string(body))
+	if len(matches) < 2 {
+		return nil, errors.New("minikube, failed to find the Kubernetes versions slice")
+	}
+
+	// Parse and cleanup the slice values
+	minikubeVersions := strings.Split(strings.NewReplacer("\n", "", `"`, "", "\t", "", " ", "").Replace(matches[1]), ",")
+
+	logDebug("Found minikube versions: %s", minikubeVersions)
+
+	var latestMinikubeVersions []string
+	// the minikube version slice is sorted, break when first cycle match is found
+	for _, k8sVersion := range k8sVersions {
+		for _, minikubeVersion := range minikubeVersions {
+			if strings.Contains(minikubeVersion, k8sVersion.Cycle) {
+				latestMinikubeVersions = append(latestMinikubeVersions, minikubeVersion)
+				break
+			}
+		}
+	}
+
+	return latestMinikubeVersions, nil
+}
+
+// getLatestSupportedKindImages iterates through the K8s supported versions and find the latest kind
+// tag that supports that version
+func getLatestSupportedKindImages(url string, k8sVersions []KubernetesVersion) ([]string, error) {
+	var supportedKindVersions []string
+	for _, k8sVersion := range k8sVersions {
+		tag := k8sVersion.Latest
+		for {
+			exists, err := imageTagExists(url, tag)
+			if err != nil {
+				return supportedKindVersions, fmt.Errorf("failed to check image tag existence: %w", err)
+			}
+			if exists {
+				supportedKindVersions = append(supportedKindVersions, "v"+tag)
+				break
+			}
+			tag, err = decrementMinorMinorVersion(tag)
+			if err != nil {
+				// It's possible that kind still does not have a tag for new versions, break the loop and
+				// process other k8s versions
+				if strings.Contains(err.Error(), "minor version cannot be decremented below 0") {
+					logDebug("No kind image found for k8s version %s", k8sVersion.Cycle)
+					break
+				}
+				return supportedKindVersions, fmt.Errorf("failed to decrement k8sVersion: %w", err)
+			}
+		}
+	}
+	return supportedKindVersions, nil
+}
+
+func imageTagExists(url string, tag string) (bool, error) {
+	body, err := getRequestBody(url + tag)
+	if err != nil {
+		return false, fmt.Errorf("failed to get image tag: %w", err)
+	}
+
+	var kindImage DockerImage
+	if err = json.Unmarshal(body, &kindImage); err != nil {
+		return false, fmt.Errorf("failed to unmarshal JSON: %w", err)
+	}
+
+	if kindImage.Count > 0 {
+		return true, nil
+	}
+	return false, nil
+}
+
+func decrementMinorMinorVersion(version string) (string, error) {
```

Comment:
It's called the patch version in semver :) 
```suggestion
func decrementPatchVersion(version string) (string, error) {
```

Reply:
roger that!